scheduler/task_finder.go

```diff
@@ -23,18 +25,54 @@ func LegacyFindRunnableTasks() ([]task.Task, error) {
 		return nil, err
 	}
 
+	projectRefCache, err := getProjectRefCache()
+	if err != nil {
+		return nil, err
+	}
+
 	// filter out any tasks whose dependencies are not met
 	runnableTasks := make([]task.Task, 0, len(undispatchedTasks))
 	dependencyCaches := make(map[string]task.Task)
 	for _, task := range undispatchedTasks {
+		ref, ok := projectRefCache[task.Project]
+		if !ok {
+			grip.Notice(message.Fields{
+				"runner":  RunnerName,
+				"message": "could not find project for task",
+				"outcome": "skipping",
+				"task":    task.Id,
+				"project": task.Project,
+			})
+			continue
+		}
+
+		if !ref.Enabled {
```

Comment:
Add a test for this?

Reply:
I'm adding more data to the test fixtures that will exercise these parts of the code.